doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -824,12 +700,10 @@ interactively, they cannot be part of a vernacular file loaded via
    discarded. In this case, :cmd:`Back` will then undo all the proof steps up to
    the statement of this proof.
 
-   .. cmdv:: Back @num
+   If :n:`@num` is specified, the command undoes :n:`@num` vernacular commands.
 
-      Undo :n:`@num` vernacular commands. As for Back, some extra
-      commands may be undone in order to reach an adequate state. For
-      instance Back :n:`@num` will not re-enter a closed proof, but rather go just
-      before that proof.
+   :cmd:`Back` will not reopen a closed proof, but instead will go to the state just before that
+   proof.
```

Comment:
By mentioning them separately, the original wording suggests that "proof management commands" are not "vernacular commands".  Strike that sentence?

Seems like `Back` and `Undo` should be in the same chapter and section.  WDYT?

The description doesn't explain this case clearly:  `Back` takes you to the state just before Proof or before Theorem?  Command2 info is no longer accessible at all after the Qed and so is not counted against @num?

```
Command1.
Theorem ..
Proof.
  : 
Command2
Qed.
Back.
```

Reply:
I agree that "This command also handles proof management commands" suggests this, and it would be good to change it to something that doesn't, like "When handling proof management commands, ..."

> Seems like `Back` and `Undo` should be in the same chapter and section. WDYT?

Yes, and they will probably belong to the "Command-line and graphical tools" chapter.

> The description doesn't explain this case clearly: `Back` takes you to the state just before Proof or before Theorem? Command2 info is no longer accessible at all after the Qed and so is not counted against `@num`?

From my tests:

```coq
Theorem foo : True.
easy.
Qed.
```

`Back` / `Back 1` take you back at the state after `Theorem`. `Back 2` takes you back at the state before `Theorem`. So yes, the commands in proof mode are not counted against `num`.